api/src/main/java/org/apache/iceberg/Transaction.java

```diff
@@ -44,6 +44,13 @@ public interface Transaction {
    */
   UpdatePartitionSpec updateSpec();
 
+  /**
+   * Create a new {@link AddPartitionSpec} to alter the partition spec of this table.
```

Comment:
This would only be able to add a new Partition Spec to the table correct? We can't actually alter any existing specs

Reply:
Correct